cryocare/protocols/protocol_training.py

```diff
@@ -18,36 +22,103 @@ def _defineParams(self, form):
+                      default=0.0004,
+                      label='Learning rate',
+                      help='Training learning rate.')
+        form.addParam('model_name', StringParam,
+                      default='model',
+                      label='Model name',
+                      help='Name of the model.')
+        form.addSection(label='U-Net Parameters')
+        form.addParam('unet_kern_size', IntParam,
+                      default=3,
+                      label='Convolution kernel size',
+                      help='Size of the convolution kernels used in the U-Net.')
+        form.addParam('unet_n_depth', IntParam,
+                      default=2,
+                      label='U-Net depth',
+                      help='Depth of the U-Net.')
+        form.addParam('unet_n_first', IntParam,
+                      default=16,
+                      label='Number of initial feature channels',
+                      help='Number of initial feature channels.')
+
 
-    # --------------------------- STEPS functions ------------------------------
     def _insertAllSteps(self):
-        # Insert processing steps
-        self._insertFunctionStep('trainStep')
+        self._insertFunctionStep('prepareTrainingStep')
+        self._insertFunctionStep('trainingStep')
         self._insertFunctionStep('createOutputStep')
 
-    def trainStep(self):
-        pass
+    def prepareTrainingStep(self):
+        config = {
+            'train_data': self.train_data.get().getTrainData(),
+            'epochs': self.epochs.get(),
+            'batch_size': self.batch_size.get(),
+            'unet_kern_size': self.unet_kern_size.get(),
+            'unet_n_depth': self.unet_n_depth.get(),
+            'unet_n_first': self.unet_n_first.get(),
+            'learning_rate': self.learning_rate.get(),
+            'model_name': self.model_name.get(),
+            'path': self._getExtraPath()
+        }
+
+        self.config_path = params.String(join(self._getExtraPath(), 'train_config.json'))
+        with open(self.config_path.get(), 'w+') as f:
+            json.dump(config, f, indent=2)
+
+    def trainingStep(self):
+        Plugin.runCryocare(self, 'cryoCARE_train.py', '--conf {}'.format(self.config_path.get()))
 
     def createOutputStep(self):
-        pass
+        model = CryocareModel(basedir=self._getExtraPath(), model_name=self.model_name.get())
+        self._defineOutputs(model=model)
+
 
     # --------------------------- INFO functions -----------------------------------
     def _summary(self):
-        """ Summarize what the protocol has done"""
         summary = []
 
         if self.isFinished():
-            summary.append("Some message to summarize.")
+            summary.append("Model saved to: {}".format(self._getExtraPath(self.model_name.get())))
         return summary
 
-    def _methods(self):
-        return []
+    def _validate(self):
+        validateMsgs = []
+
+        if not self.train_data.get():
```

Comment:
You can use Validators at defineParams: There are Positive validators and more:
at pyworkflow/protocol/params.py